### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by TamaraDimov**
-Implement a function for adding a new task (add a new element to the array).
-Implement a function for deleting a task (remove an element from the array).
-Implement a function for editing task descriptions.
-update read me

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,38 +1,128 @@
...
+  }
+  return toDoListArray;
+};
+
+const toDoTasks = getLocalStorage();
+const toDoList = document.querySelector('.toDoContainer');
+
+const addTask = (toDoListArray, task) => {
+  toDoListArray.push({ task, id: toDoListArray.length + 1 });
+};
```
Comment (cynthiainga) [inline src/index.js]: - [ ] When adding a task to todo-list by default, new tasks should have the property completed set to false, but in your case when adding a task to the list the property complete is not showing, we need to check the box to see the complete if is true or false, but it should be set to false by default and show directly in the localStorage when adding a task to the list. Please kindly fix that 🙃🙃

Screen Shot 2023-01-19 at 15 40 01

File: src/index.js [hunk 2]
```diff
@@ -1,38 +1,128 @@
...
+
+    toDoList.appendChild(toDoItem);
+  });
+};
+
+const editTask = (e) => {
+  const clickedTask = e.target.closest('.toDoContainer-li-text');
+  clickedTask.disabled = false;
+  clickedTask.focus();
+};
```
Comment (cynthiainga) [inline src/index.js]: - [ ] When I edit a task, the change doesn’t save in the localStorage, cause after editing a task all the changes disappear if I refresh the page. Please kindly fix that as well 🙃😉

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @TamaraDimov, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] No linters errors ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Correct Github flow ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇 👇 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 3]
```diff
@@ -1,38 +1,142 @@
...
+    if (clickedTask.value !== '') {
+      const taskIndex = toDoListArray.findIndex(
+        (task) => task.task === taskText,
+      );
+      toDoListArray[taskIndex].task = clickedTask.value;
+      clickedTask.disabled = true;
+      updateLocalStorage(toDoListArray);
+      renderToDoList(toDoListArray);
+    }
+  });
```
Comment (tresorsawasawa) [inline src/index.js]: - [ ] Good job trying to fix some of the changes required by the previous reviewer. While fixing the Editing functionality you introduced a new error for editing a task as now it only accepts one letter.
You can see below how your app behaviors:
edit

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
.

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @TamaraDimov 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**